core/chains/evm/log/orm.go

```diff
@@ -60,16 +60,18 @@ func NewORM(db *sqlx.DB, lggr logger.Logger, cfg pg.LogConfig, evmChainID big.In
 	return &orm{pg.NewQ(db, lggr, cfg), *utils.NewBig(&evmChainID)}
 }
 
-func (o *orm) WasBroadcastConsumed(blockHash common.Hash, logIndex uint, jobID int32, qopts ...pg.QOpt) (consumed bool, err error) {
+func (o *orm) WasBroadcastConsumed(blockHash common.Hash, txIndex uint, logIndex uint, jobID int32, qopts ...pg.QOpt) (consumed bool, err error) {
 	query := `
 		SELECT consumed FROM log_broadcasts
 		WHERE block_hash = $1
-		AND log_index = $2
-		AND job_id = $3
-		AND evm_chain_id = $4
+		AND (tx_index is NULL OR tx_index = $2) -- NULL-check only for safe upgrade from 1.6.0
```

Comment:
Do we need to clean up `NULL` `tx_index` columns along the way?
Consider if one (`A`) of two previously conflicting logs (`A`,`B`) was consumed, and then both are replayed after upgrading. This check would match both `A` and `B`, which results in `B` never actually being consumed.

Is it possible to fill in these null columns with the value from the first log? (is that even an invariant that holds? or could it be any of them that was actually consumed?)